docs/overview/architecture.md

```diff
@@ -0,0 +1,245 @@
+9. If Vald Agent successfully inserts the request data, it will return success to the Vald LB Gateway.
+10. After Vald LB Gateway receives success from the selected Vald Agents, it will respond the IP addresses of all selected Vald Agents to the Vald Backup Gateway.
+11. Vald Backup Gateway returns success to Vald Meta Gateway.
+12. Vald Meta Gateway will forward the UUID(s) and vector ID(s) to the Vald Meta.
+13. Vald Meta will store the UUID(s) and vector ID(s) that were successfully processed by the Vald Agent(s) to the persistent layer such as Redis, Cassandra, MySQL, etc.
+14. Vald Backup Gateway will asynchronously send all the inserted the data (vector(s), vector ID(s), UUID(s) and IP address(es)) to the Vald Compressor. Vald Compressor will compress the vector data asynchronously to reduce the size of the vector data.
+15. Vald Compressor will forward the data (compressed vector(s), vector ID(s), UUID(s) and IP address(es)) to the Vald Backup Manager.
+16. Vald Backup Manager will store all of the data to the persistent layer such as MySQL, Cassandra, etc., to prevent the data lost in Vald.
+17. Vald Meta Gateway will return success to the Vald Filter Gateway.
+18. Vald Filter Gateway will return success to the Vald Ingress.
+
+### Search
+
+<img src="../../assets/docs/search_flow.svg" />
+
+When the user searches a vector from Vald:
+
+1. Vald Ingress receives a search request from the user. Vald provides 2 searching interfaces to the user, the user can search by vector or search by the vector ID.
+2. Vald Ingress will forward the request to the Vald Filter Gateway to pre-process the request data.
+3. Vald Filter Gateway will forward the request to the user-defined Vald Ingress Filter. After the Vald Ingress Filter received the request, it will perform the pre-processing logic defined by the user, for example, padding the vector to match the vector dimension in Vald.
+4. After the request is processed by the user-defined Vald Ingress Filter, the result will return to the Vald Filter Gateway.
+5. Vald Filter Gateway will forward the request to the Vald Meta Gateway. Vald Meta Gateway is used to resolve the internal used UUID to the user inserted vector ID in step 10-11.
+6. Vald Meta Gateway will forward the request to the Vald LB Gateway. Vald LB Gateway will preform the post-processing of the result in step 9 after the Vald Agent(s) return in step 8.
+7. Vald LB Gateway will forward the request to all Vald Agents in parallel. Each Vald Agent will search the nearest vectors in an on memory cache.
+8. Vald Agent returns the searching result to the Vald LB Gateway. The searching result includes the UUID, the vector distance, and the vector. The number of the result will be the same as requested.
+9. After all Vald Agents return the searching result, Vald LB Gateway will aggregate all of the results and reorder the result by the vector distance, and return the reordered result to the Vald Meta Gateway.
```

Comment:
- I'm not sure `reorder` is the most suited word in this sentence... Where to Vald LB Gateway reorder?

Reply:
I think reorder means relank search result vector:thinking:

Reply:
I see your consideration, and I think `reoreder` is not suit and not correct in this sentence.
User may think `reoreder` means "Vald Meta Gateway request to Vald Agents again in 9".

Reply:
```suggestion
9. Vald LB Gateway will aggregate all searching results from all Vald Agents, rank the result by the vector distance, and return the ranked result to the Vald Meta Gateway.
```

Reply:
Thank you for your changes.
I think it is good.